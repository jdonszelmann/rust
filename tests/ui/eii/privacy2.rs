//@ aux-build:codegen3.rs
#![feature(eii)]

extern crate codegen3 as codegen;

#[codegen::eii1]
fn eii1_impl(x: u64) {
    println!("{x:?}")
}

#[codegen::eii3] //~ ERROR failed to resolve: could not find `eii3` in `codegen`
fn eii3_impl(x: u64) {
    println!("{x:?}")
}

// what you would write:
fn main() {
    // directly
    eii1_impl(21);
    // through the alias
    codegen::decl1(42); //~ ERROR function `decl1` is private
}
