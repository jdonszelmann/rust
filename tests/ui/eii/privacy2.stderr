error[E0433]: failed to resolve: could not find `eii3` in `codegen`
  --> $DIR/privacy2.rs:15:12
   |
LL | #[codegen::eii3]
   |            ^^^^ could not find `eii3` in `codegen`

error[E0603]: function `decl1` is private
  --> $DIR/privacy2.rs:25:14
   |
LL |     codegen::decl1(42);
   |              ^^^^^ private function
   |
note: the function `decl1` is defined here
  --> $DIR/auxiliary/codegen3.rs:7:1
   |
LL | fn decl1(x: u64);
   | ^^^^^^^^^^^^^^^^^

error: couldn't find an implementation for `#[eii2]`
  --> $DIR/auxiliary/codegen3.rs:9:1
   |
LL | #[eii(eii2)]
   | ^^^^^^^^^^^^ an implementation was expected by this declaration
   |
   = help: expected at least one implementation in crate `privacy2` or any of its dependencies

error: couldn't find an implementation for `#[eii3]`
  --> $DIR/auxiliary/codegen3.rs:13:5
   |
LL |     #[eii(eii3)]
   |     ^^^^^^^^^^^^ an implementation was expected by this declaration
   |
   = help: expected at least one implementation in crate `privacy2` or any of its dependencies

error: aborting due to 4 previous errors

Some errors have detailed explanations: E0433, E0603.
For more information about an error, try `rustc --explain E0433`.
